src/ui/flightdisplay/FlightDisplay.qml

```diff
@@ -48,65 +52,141 @@ Rectangle {
         return value ? "1" : "0";
     }
 
+    function adjustSizes() {
+        var dist = 85
+        var wide = 160
+        var minw = 496
+        if(root.width > minw)
+        {
+            attitudeInstrument.size = wide;
+            attitudeInstrument.x    = dist
+            compassInstrument.size  = wide;
+            compassInstrument.x     = root.width - wide - dist
+        } else {
+            var factor = (root.width / minw);
+            var ndist  = dist * factor;
+            var nwide  = wide * factor;
+            if (ndist < 0)
+                ndist = 0;
+            attitudeInstrument.size = nwide;
+            compassInstrument.size  = nwide;
+            attitudeInstrument.x    = ndist;
+            compassInstrument.x     = root.width - nwide - ndist;
+        }
+    }
+
     Component.onCompleted:
     {
         mapBackground.visible               = getBool(flightDisplay.loadSetting("showMapBackground",        "0"));
         mapBackground.alwaysNorth           = getBool(flightDisplay.loadSetting("mapAlwaysPointsNorth",     "0"));
-        attitudeWidget.visible              = getBool(flightDisplay.loadSetting("showAttitudeWidget",       "1"));
-        attitudeWidget.displayBackground    = getBool(flightDisplay.loadSetting("showAttitudeBackground",   "1"));
-        pitchWidget.visible                 = getBool(flightDisplay.loadSetting("showPitchWidget",          "1"));
+        showAttitudeIndicator               = getBool(flightDisplay.loadSetting("showAttitudeIndicator",    "1"));
+        showPitchIndicator                  = getBool(flightDisplay.loadSetting("showPitchIndicator",       "1"));
+        showCompass                         = getBool(flightDisplay.loadSetting("showCompass",              "1"));
         altitudeWidget.visible              = getBool(flightDisplay.loadSetting("showAltitudeWidget",       "1"));
         speedWidget.visible                 = getBool(flightDisplay.loadSetting("showSpeedWidget",          "1"));
-        compassIndicator.visible            = getBool(flightDisplay.loadSetting("showCompassIndicator",     "1"));
         currentSpeed.showAirSpeed           = getBool(flightDisplay.loadSetting("showCurrentAirSpeed",      "1"));
         currentSpeed.showGroundSpeed        = getBool(flightDisplay.loadSetting("showCurrentGroundSpeed",   "1"));
         currentAltitude.showClimbRate       = getBool(flightDisplay.loadSetting("showCurrentClimbRate",     "1"));
         currentAltitude.showAltitude        = getBool(flightDisplay.loadSetting("showCurrentAltitude",      "1"));
         mapTypeMenu.update();
-    }
-
-    Rectangle {
-        id: windowBackground
-        anchors.fill: parent
-        anchors.centerIn: parent
-        visible: !attitudeWidget.visible && !mapBackground.visible
-        color:   Qt.hsla(0.25, 0.5, 0.45)
-        z:       0
+        adjustSizes();
     }
 
     Menu {
```

Comment:
Is there a font for Menus? If so we probably need a QGCMenu and friends to pick up the default font.